flink-runtime/src/main/java/org/apache/flink/runtime/deployment/ResultPartitionDeploymentDescriptor.java

```diff
@@ -59,12 +61,42 @@ public ResultPartitionDeploymentDescriptor(
 			ShuffleDescriptor shuffleDescriptor,
 			int maxParallelism,
 			boolean sendScheduleOrUpdateConsumersMessage) {
+		this(
+			partitionDescriptor,
+			shuffleDescriptor,
+			maxParallelism,
+			sendScheduleOrUpdateConsumersMessage,
+			// Later we might have to make the scheduling adjust automatically
+			// if certain release type is not supported by shuffle service implementation at hand
+			partitionDescriptor.getPartitionType() != ResultPartitionType.BLOCKING);
```

Comment:
I think in future the `releasedOnConsumption` could be refactored into one `ReleaseStrategy` as proposed in #8804 , then it could be configured which strategy is used in practice and check whether this config is consistent with shuffle implementation.

Reply:
True, the PR just leaves it as it was before at the moment